_posts/2021-07-21-documentation-survey.md

```diff
@@ -0,0 +1,177 @@
+
+<img src="/images/cudf-docs.png" width="70%">
+
+Tutorials, API and some explanations of internals are intermingled without a clear hierarchy.
+
+### [xarray](http://xarray.pydata.org/en/stable/)
+
+There is no top-level nav at all and the left hand nav is split into "For Users", "For Developers/Contributors" and "Community".
+
+<img src="/images/xarray-docs.png" width="70%">
+
+There are several subsections within "For Users" that have their own dropdowns such as "Getting Started" which has "Overview: Why xarray?", "Installation", "Quick overview" and "Frequently Asked Questions"
+
+There is also this "How do I ..." section:
+
+<img src="/images/xarray-how-do-I.png" width="70%">
+
+## R libraries
+
+R has the concept of "vignettes" which offer narrative style introductions to R package. In addition each package has a pdf that is essentially an API Reference.
+
+### [dplyr](https://dplyr.tidyverse.org/)
+
+The landing page contains most of the contents. Including an overview that includes top-level function definitions, installation information, some usage examples, and links for getting in touch.
+
+The most prominent links are to "Get Started", "Reference", and "Articles". "Get Started" links to the vignette which introduces concepts while working through a toy problem. "Reference" contains links to all functions grouped by rough use-case and development status.
+
+<img src="/images/r-dplyr-reference.png" width="70%">
+
+"Articles" links to something more like a traditional "User Guide" which provides explanation and more detail about particular methods.
+
+## ML libraries
+
+### [PyTorch](https://pytorch.org/docs/stable/index.html)
+
+The PyTorch docs feature the Python API very prominently, and also contain implementation notes. The "Get Started" and "Tutorial" are entirely separated from the "Docs", but are accessible from the top nav. "Get Started" uses tabs effectively to separate out Local vs Remote solutions.
+
+<img src="/images/pytorch-docs.png" width="70%">
+
+The Tutorial features "Introduction to PyTorch" and also contains content-area specific Tutorials such as "Audio" and "Text". It looks like this is an aggregation where individual contributors submit tutorials with keyword tags.
+
+### [TensorFlow](https://www.tensorflow.org/overview)
+
+TensorFlow takes a similar approach of totally separating the API docs from the "Tutorials" and has an additional concept of a "Guide" which provides explanations and context. Both "Tutorials" and "Guide" are under the category of "Learn". "Learn" has an overview page that contains two self-contained hello world examples and has text explaining how to navigate the docs.
+
+<img src="/images/tensorflow-tutorials.png" width="70%">
+
+The Tutorials cover the same material as the PyTorch ones, but make the split between Beginner and Advanced with separate quickstarts for each type of user.
+
+
+## Takeaways
+
+- Don't do dropdowns on the top nav bar
+- Use the right margin for within-page nav
```

Comment:
Will this work well across all screen sizes, or restrict functionality to just the people who are working on a large screen? 
I'd imagine mobile devices are not super common among people visiting the docs (are there analytics on that?), but people out & about working off a small laptop screen are likely to be reasonably common.

Reply:
I think most projects just don't show this TOC on small screens. That seems reasonable to me since it is just the section headers.